data/dataset.py

```diff
@@ -117,15 +122,40 @@ def add_cls_token(self, src_token_ids: list[int], tgt_token_ids: list[int]):
         cls_token_id = self.tgt_encoder.token_to_id["<CLS>"]
         src_token_ids.insert(0, cls_token_id)
         tgt_token_ids.insert(0, cls_token_id)
-        
+
         return src_token_ids, tgt_token_ids
 
 
+def shard_and_build(
+    dataset: Dataset,
+    dataset_name: str,
+    dataset_cfg: DictConfig,
+    dataset_cache_path: str,
+    num_shards: int = 2,
+) -> Dataset:
+    shard_paths = []
+    for shard_i in range(num_shards):
+        path = f"{dataset_cache_path}-part-{shard_i}"
+        dataset_shard = dataset.shard(num_shards=num_shards, index=shard_i)
+        print(f"Processing shard {shard_i + 1} of {num_shards} with {len(dataset_shard)} records.")
```

Comment:
```suggestion
    for it in range(num_shards):
        path = f"{dataset_cache_path}-part-{it}"
        dataset_shard = dataset.shard(num_shards=num_shards, index=it)
        print(f"Processing shard {it} of {num_shards} with {len(dataset_shard)} records.")
```

That's my personal preference for naming for loop range iterators :+1:, and we shouldn't switch to 1-based indexing just for prints